server/src/main/java/org/opensearch/search/backpressure/stats/SearchShardTaskStats.java

```diff
@@ -30,21 +31,29 @@
 public class SearchShardTaskStats implements ToXContentObject, Writeable {
     private final long cancellationCount;
     private final long limitReachedCount;
+    private final Long completionCount;
```

Comment:
Why is this a nullable `Long`? 

If we're on or after 3.0, aren't we guaranteed to get a value in the constructor? 

The only way we won't get a value is if we're deserializing a value from an earlier version, which should only ever be a temporary state.

Reply:
Yes, you are right. I made this as `Long` instead of primitive `long` type because I had seen a comment on this by andross here: https://github.com/opensearch-project/OpenSearch/pull/7445#discussion_r1199296121 .

Reply:
Hmm... this feels like the wrong approach to me, since it's optimizing for the temporary state. In the long run, this value won't be optional. It will definitely be available.

The only time that you would create an instance of `SearchShardTaskStats` that needs a null value is if you are reading a response from an old node that predates this change, which will only happen during a rolling upgrade. But once we allow `null` as a possible value, we need to propagate it in the unlikely event that a new node deserializes a response from an old node and forwards it to another new node.

If we go with `writeOptionalLong` and `readOptionalLong`, the value needs to remain optional (forever?) to maintain backwards compatibility. We'll constantly be writing a `true` value before the `long` value. Yuck 😛 

IMO, a (temporary) sentinel -1 value when reading from an old node is preferable to adopting optional semantics for something required.

Reply:
I agree with this, I will make this change. I didn't get the following from your comment 
> We'll constantly be writing a true value before the long value. 
Can you elaborate a little more on this ?